geopandas/base.py

```diff
@@ -4104,6 +4104,103 @@ def sample_points(self, size, method="uniform", seed=None, **kwargs):
+        edges of a planar graph. Any geometry type may be provided as input; only the
+        constituent lines and rings will be used to create the output polygons.
+
+        Lines or rings that when combined do not completely close a polygon will result
+        in an empty GeoSeries. Duplicate segments are ignored.
+
+        Unless you know that the input GeoSeries represent clean topological of a planar
+        graph (e.g. there is a node on both lines where they intersect), it is
+        recommended to use ``node=True`` which performs noding prior polygonization.
+        Using ``node=False`` will provide performance benefits but may result in
+        incorrect polygons if the input is not of the proper topology.
+
+        When ``full=True``, return not only the polyoginal result all extra outputs as
+        well. The return value consists of 4 elements:
+
+            - GeoSeries of the valid polygons (same as with ``full=False``)
+            - GeoSeries of cut edges: edges connected on both ends but not part of
+              polygonal output
+            - GeoSeries of dangles: edges connected on one end but not part of polygonal
+              output
+            - GeoSeries of invalid rings: polygons that are formed but are not valid
+              (bowties, etc)
+
+        Parameters
+        ----------
+        node : bool, optional
+            perform noding prior polygonization, by default True
+        full : bool, optional
+            return the full output composed of a tuple of GeoSeries, by default False
+
+        Returns
+        -------
+        GeoSeries | tuple(GeoSeries, GeoSeries, GeoSeries, GeoSeries)
+            GeoSeries with the polygons or a tuple of four GeoSeries as
+            ``(polygons, cuts, dangles, invalid)``
+
+        Examples
+        --------
+        >>> from shapely.geometry import LineString
+        >>> s = geopandas.GeoSeries([
+        ...     LineString([(0, 0), (1, 1)]),
+        ...     LineString([(0, 0), (0, 1), (1, 1), (1, 0), (0, 0)]),
+        ... ])
+        >>> s.polygonize()
+        0    POLYGON ((0.00000 0.00000, 1.00000 1.00000, 1....
+        1    POLYGON ((1.00000 1.00000, 0.00000 0.00000, 0....
+        Name: polygons, dtype: geometry
+
+        >>> polygons, cuts, dangles, invalid = s.polygonize(full=True)
+        """
+        from .geoseries import GeoSeries
+        from .array import from_shapely
+
+        if compat.USE_SHAPELY_20:
+            from shapely import polygonize, polygonize_full
+
+        elif compat.USE_PYGEOS:
+            from pygeos import polygonize, polygonize_full
+
+        else:
+            from shapely.ops import polygonize, polygonize_full
+
+        if node:
+            geometry_input = from_shapely([self.geometry.unary_union])
+        else:
+            geometry_input = self.geometry.values
+
+        if compat.USE_PYGEOS:
+            geometry_input = geometry_input.data
```

Comment:
Should this be `_data` so we don't bubble up the pygeos access warning doesn't see the associated warning from an internal usage?
```suggestion
            geometry_input = geometry_input._data
```